### PR TITLE
feat: add HTML reporter for visual diff dashboard

### PR DESCRIPTION
**Pull request by pftg**
## Summary

Interactive HTML report generated with zero configuration:

```ruby
# test_helper.rb — one line
require 'capybara_screenshot_diff/reporters/html'
```

After tests: `open tmp/csd-report/index.html`

## Features
- Side-by-side baseline vs current comparison
- Diff toggle button (switch between actual and annotated diff)
- Searchable sidebar with thumbnails
- Summary stats bar (total/failed/passed/success rate)
- Zero config — `at_exit` hook auto-generates on require (SimpleCov pattern)
- Works with Minitest, RSpec, and Cucumber — no framework-specific code

## Architecture
- Reporter registers via `CapybaraScreenshotDiff.reporters` array
- `reset` notifies reporters before clearing assertions (captures data before teardown)
- `at_exit` calls `finalize` to write the HTML file
- Template based on prototype from showcase project (Tailwind CSS + Alpine.js)

## Test plan
- [x] 4 tests: empty assertions, passing only, failing generates HTML, summary stats
- [x] Full suite with CI=true (200 runs, 0 failures)
- [x] Pre-PR: brutal review (1.5, fixed rescue) + codex review (P1 assertion timing fixed)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by Sourcery

Add an HTML reporter that generates an interactive visual diff dashboard for failed screenshot comparisons and integrates it into the screenshot assertion lifecycle.

New Features:
- Introduce a pluggable HTML reporter that aggregates screenshot comparison results and writes an HTML report with summary statistics and thumbnails.
- Render a Tailwind/Alpine-powered HTML dashboard template for browsing failed visual diffs with side-by-side and diff views.
- Expose a reporters collection on CapybaraScreenshotDiff to allow reporters to be registered and notified of assertion results.

Enhancements:
- Update the screenshot assertion reset flow to notify registered reporters of collected assertions before clearing them, with robust error handling.
- Document the HTML report usage and entrypoint in the README so it can be enabled with a single require in test setup.

Tests:
- Add unit tests covering HTML reporter behavior for empty, passing-only, and mixed assertion sets, including HTML generation and summary statistics.

## Summary by CodeRabbit

* **New Features**
  * Interactive HTML report for screenshot test differences with summary stats, searchable thumbnails, and a toggle to view baseline vs diff images.

* **Documentation**
  * README instructions for enabling and opening the HTML screenshot-diff report.

* **Tests**
  * Unit tests covering report generation, summaries, and output behavior.

* **Bug Fixes**
  * Report generation is resilient to problematic comparison data and continues producing the report when possible.

### CODE REVIEW

File: lib/capybara_screenshot_diff/reporters/templates/report.html.erb [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+      on: false,
+      toggle() { this.on = !this.on }
+    })
+
+    Alpine.store('screenshots', {
+      current: 0,
+      search: '',
+      items: <%= failed_screenshots.to_json %>,
+
+      init() {
```
[the remark this answers is not in the document]
Comment (pftg) [reply lib/capybara_screenshot_diff/reporters/templates/report.html.erb]: Fixed: removed `init()`, now calling `store.select(0)` after store creation.

File: lib/capybara_screenshot_diff/reporters/html.rb [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+  end
+end
+
+CapybaraScreenshotDiff.reporters << CapybaraScreenshotDiff::Reporters::HTML.new
+
+at_exit do
+  CapybaraScreenshotDiff.reporters.each(&:finalize)
+rescue => e
+  warn "[capybara-screenshot-diff] HTML report generation failed: #{e.message}"
+end
```
[the remark this answers is not in the document]
Comment (pftg) [reply lib/capybara_screenshot_diff/reporters/html.rb:91]: Fixed: now logs `e.class` and `e.full_message`.

File: test/unit/reporters/html_reporter_test.rb [hunk 3]
```diff
@@ -0,0 +1,81 @@
+# frozen_string_literal: true
+
+require "test_helper"
+require "capybara_screenshot_diff/reporters/html"
+
+module CapybaraScreenshotDiff
+  module Reporters
+    class HTMLReporterTest < ActiveSupport::TestCase
```
[the remark this answers is not in the document]
Comment (pftg) [reply test/unit/reporters/html_reporter_test.rb:8]: Fixed: added error-tolerance test with broken assertion.

File: test/unit/reporters/html_reporter_test.rb [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+      test "#format includes summary stats" do
+        reporter = HTML.new(output_path: @output_path)
+
+        assertions = [
+          build_failing_assertion("page_a"),
+          build_passing_assertion("page_b"),
+          build_failing_assertion("page_c")
+        ]
+        reporter.format(assertions)
+
```
[the remark this answers is not in the document]
Comment (pftg) [reply test/unit/reporters/html_reporter_test.rb]: Fixed: using regex `/Total.*<strong>3/` instead of bare string match.

File: lib/capybara_screenshot_diff/reporters/html.rb [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+          compare = assertion.compare
+          next unless compare
+
+          @total += 1
+          next unless compare.difference&.different?
+
+          @all_failures << build_failure_entry(assertion)
+        rescue => e
+          warn "[capybara-screenshot-diff] Reporter skipped '#{assertion.name}': #{e.message}"
+        end
```
[the remark this answers is not in the document]
Comment (pftg) [reply lib/capybara_screenshot_diff/reporters/html.rb:31]: Acknowledged as v1 limitation. Thread safety documented in README. The assertion registry is already thread-local, so in practice each thread's assertions stay isolated — the reporter just accumulates from whatever thread calls reset.

File: lib/capybara_screenshot_diff/reporters/html.rb [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+    end
+  end
+end
+
+CapybaraScreenshotDiff.reporters << CapybaraScreenshotDiff::Reporters::HTML.new
+
+at_exit do
+  CapybaraScreenshotDiff.reporters.each(&:finalize)
+rescue => e
+  warn "[capybara-screenshot-diff] HTML report generation failed: #{e.message}"
```
[the remark this answers is not in the document]
Comment (pftg) [reply lib/capybara_screenshot_diff/reporters/html.rb]: Documented in README: 'In parallel test environments, each worker writes to the same file.' This is a v1 limitation — consistent with SimpleCov's behavior.

File: README.md [hunk 7]
```diff
@@ -736,6 +736,23 @@ Capybara::Screenshot.capybara_screenshot_options[:full_page] = true
...
+```
+
+After running tests, open the report:
+
+```bash
+open tmp/csd-report/index.html
+```
+
+The report includes a sidebar with thumbnails, side-by-side comparison with diff toggle, search, and summary stats. No configuration needed — just require it.
+
```
[the remark this answers is not in the document]
Comment (pftg) [reply README.md]: Fixed: README now states 'generated only when there are failures'.

File: test/unit/reporters/html_reporter_test.rb [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+        html = @output_path.read
+        assert_includes html, "3"  # total
+        assert_includes html, "2"  # failed
+      end
```
[the remark this answers is not in the document]
Comment (pftg) [reply test/unit/reporters/html_reporter_test.rb]: Fixed: using regex matching against labeled summary values.
